### PR TITLE
`<SiteThumbnail>`: onload handling doesn't fire once component has unmounted

### PR DESCRIPTION
**Pull request by p-jackson**
#### Proposed Changes

A warning in React said the `onload` handler was still being called after the component had unmounted.

* Remove `ref` in favor of image props
* Fixes the mShots on Safari

#### Testing Instructions

* Go to `/sites` in Safari browser
* Change between row and tile views
* Observe the mShots appear consistently
* Open the console, observe no warning or errors appear related to react memory leaks

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

- Related to https://github.com/Automattic/wp-calypso/issues/67202

### CODE REVIEW

File: packages/components/src/site-thumbnail/use-mshots-img.tsx [hunk 1]
```diff
@@ -27,52 +27,55 @@ export const useMshotsImg = (
...
-	}, [ imgRef, mshotUrl, retryCount ] );
+		clearTimeout( timeout.current );
+	}, [ mshotUrl ] );
 
 	return {
-		src: mshotUrl,
 		isLoading,
 		isError,
-		imgRef,
+		imgProps: { onLoad, onError, src: mshotUrl, loading: 'lazy' },
```
Comment (sejas) [inline packages/components/src/site-thumbnail/use-mshots-img.tsx:79]: I like using `imgProps` instead of `imgRef`

**Review by sejas — APPROVED** [review]
It's looking good 🥳 
Much better using `props` instead of `ref` 👍

File: packages/components/src/site-thumbnail/use-mshots-img.tsx (hunk 1 above)
Comment (vykes-mac) [reply packages/components/src/site-thumbnail/use-mshots-img.tsx:79]: This is a nice alternative. Removing the onload event in the useEffect cleanup function could also solve this issue ?

Comment (p-jackson) [reply packages/components/src/site-thumbnail/use-mshots-img.tsx:79]: Sorry I haven't written the PR description yet, but the `onload` handler is what alerted me to the issue. When quickly switching from grid to table view I got a React error in the console about setting state on an unmounted component. The problem is that the `onload` handlers for previous thumbnail components are being called after we've made the switch.

Usually the `onload` handler wouldn't get called once the `` element has been removed from the DOM. So something is holding onto a reference to that element and keeping it alive.

And that seems to be the `imgRef`. I think using it in the `useEffect` dependency list is what does it. Using refs this way (where you have a hook create the ref, and then returning it to be used by someone else) seems rife for memory leaks. So I like the idea of only returning the callbacks. Ownership is less ambiguous; only the `<SiteThumbnail>` component ever has ownership over the `` element.

**Pull request merged**